### PR TITLE
@sweir27: Document deployments

### PR DESCRIPTION
**Pull request by joeyAghion**
This new playbook is based on the overview of deployment I hosted for onboarding/re-indoctrination. It could go under the "onboarding sessions" you started, but I tried to flesh it out enough to stand alone as a self-service reference so think it might qualify as a playbook.

### CODE REVIEW

File: playbooks/deployments.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+- _Safe_: Only successfully-tested code should be deployed. Deploying should be _graceful_. I.e., shouldn't impact
+  availability of the service.
+- _Repeatable_: The result of a deploy should be consistent no matter the state of the developer's local
+  environment, the particulars of a code change, the state of data, volume of traffic, etc.
+- _Visible_: It should be transparent what deploy is happening when (and by whom...). An audit trail and
+  instrumentation can be helpful in diagnosing production issues and interpreting monitoring results.
+- _Simple_: Other things being equal, we prefer simple deployment operations whose impact can be easily understood
+  by developers who may have to consider how to undo a change or its impact on other systems.
+- _Reversible\*_: We should be able to roll back a deploy while keeping the state of the system stable and
```
Comment (sweir27) [inline playbooks/deployments.md]: What's the `\*` for?

File: playbooks/deployments.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  consistent. (This may also include associated changes such as to dependencies or configuration.)
+- _Consistent\*_: Ideally the deployment operation is the same whether a change includes just code, updated
+  configuration, data migrations, or dependencies.
+- _Fast\*_: Other things being equal, we prefer deploys to be fast so we can deploy often and focus on other
```
Comment (sweir27) [inline playbooks/deployments.md]: Do you mean fast as in "it should not take long for the deploy script to finish"?

One thing that seems to be missing from this list is high frequency-- we should encourage changes to be deployed to production as soon as they can be (early and often?).

File: playbooks/deployments.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  ([Kaws](https://github.com/artsy/kaws/blob/2e8e0ca8be2bc14e30e979cf2481e01767762f5e/.circleci/config.yml),
+  [Force](https://github.com/artsy/force/blob/4e81b8f92d40bcdcd6c575be31ac561e1500f203/.circleci/config.yml))
+- via `git push heroku master` ([Impulse](https://github.com/artsy/impulse/blob/master/.circleci/config.yml))
+
+This leverages the exact set-up already in use to deploy staging, and is independent of any developer's
+environment. The pull request also acts as a convenient record of the deploy, including commits and any discussion.
+It's possible, however, for deploys to end up blocked or delayed by other CI tasks. This approach also requires a
+little extra CircleCI configuration.
+
+Some projects trigger a deploy by PR-ing `master` to a `release` branch. This risks including commits that haven't
```
Comment (sweir27) [inline playbooks/deployments.md]: 🙏 many thanks for explaining this here.

File: playbooks/deployments.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+completed a full CI run over the `master` branch. To avoid this, others update a `staging` or `master-stable`
+branch after the staging deploy is complete and PR from that branch.
+
+## Recommendations
+
+- It's acceptable for projects to start by recommending a basic `hokusai pipeline promote`
+- When setting up projects, use `hokusai setup` in combination with the templates in
+  [artsy/artsy-hokusai-templates](https://github.com/artsy/artsy-hokusai-templates). They provide reasonable
+  starting points for `.circleci/config.yml` and other configuration files.
+- Full-fledged projects should drive releases (via CircleCI) with pull requests from a `staging` branch to a
```
Comment (sweir27) [inline playbooks/deployments.md]: There's been an issue with people having chrome extensions that delete git branches that cause them to accidentally delete `staging` if they run a deploy. I wonder if it's worth mentioning here (maybe not).

File: playbooks/deployments.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    it's possible that doesn't match what was merged into the `release` branch by the PR. In the future, we'd like
+    to avoid this race condition with an argument to `hokusai pipeline promote`, possibly like
+    `--only $CIRCLE_SHA1`.
+
+In all cases:
+
+- Announce deploys for visibility in [#dev](https://artsy.slack.com/messages/dev)
+- Document accurate deploy procedure in README
+
+## Advanced topics
```
Comment (sweir27) [inline playbooks/deployments.md]: Can we add rollbacks to here?

Comment (sweir27) [reply playbooks/deployments.md]: Oh just had a thought-- maybe on head project's README we should _also_ describe how to roll back a deploy.

File: playbooks/deployments.md (hunk 1 above)
Comment (joeyAghion) [reply playbooks/deployments.md]: Updated to explain.

File: playbooks/deployments.md (hunk 2 above)
Comment (joeyAghion) [reply playbooks/deployments.md]: Good point. It's more about "fresh" and "granular" than "frequent" though. (A stable system may only be deployed infrequently, but should still be deployable shortly after any change.) I tweaked the description a bit.

File: playbooks/deployments.md (hunk 4 above)
Comment (joeyAghion) [reply playbooks/deployments.md]: Tell me more...

It's common for repos to have long-running branches be the base of PRs (e.g., a `develop` branch in some flows), so such an extension would seem to be very dangerous.

File: playbooks/deployments.md (hunk 5 above)
Comment (joeyAghion) [reply playbooks/deployments.md]: Updated to mention roll-backs in hot-fixes.

File: playbooks/deployments.md (hunk 4 above)
Comment (sweir27) [reply playbooks/deployments.md]: i.e. https://artsy.slack.com/archives/C0C4AJ1PF/p1499794598511441

**Pull request merged**